src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -92,12 +96,14 @@ class ShenandoahPurgeSATBTask : public AbstractGangTask {
   ShenandoahObjToScanQueueSet* _mark_queues;
 
 public:
-  size_t _trashed_oops;
+  volatile size_t _trashed_oops;
```

Comment:
If it the intention is to make shared access to the field safe, then C++ volatile does not (fully) achieve this. You also need to use Atomic::inc() and similar to change/read it.

Reply:
Adding volatile keyword, mainly just for coding convention on shared variable. It uses Atomic::add() to update variable, so it is correct.